lib/smart_answer/calculators/coronavirus_find_support_calculator.rb

```diff
@@ -14,14 +14,45 @@ class CoronavirusFindSupportCalculator
                   :have_you_been_made_unemployed,
                   :mental_health_worries
 
-    def has_results?
-      return false if need_help_with.blank?
+    RULES = {
+      feeling_unsafe: lambda { |calculator|
+        calculator.needs_help_with?("feeling_unsafe") && calculator.feel_safe != "yes"
+      },
+      paying_bills: lambda { |calculator|
+        calculator.needs_help_with?("paying_bills") && calculator.afford_rent_mortgage_bills != "no"
+      },
+      getting_food: lambda { |calculator|
+        calculator.needs_help_with?("getting_food") &&
+          (calculator.afford_food != "no" && calculator.get_food != "yes")
+      },
+      going_to_work: lambda { |calculator|
+        calculator.needs_help_with?("going_to_work") && calculator.worried_about_work != "no"
+      },
+      being_unemployed: lambda { |calculator|
+        calculator.needs_help_with?("being_unemployed") && (
+          (calculator.self_employed != "yes" && calculator.have_you_been_made_unemployed != "no") || calculator.are_you_off_work_ill == "yes"
+        )
+      },
+      somewhere_to_live: lambda { |calculator|
+        calculator.needs_help_with?("somewhere_to_live") && (
+          calculator.have_somewhere_to_live != "yes" || calculator.have_you_been_evicted != "no"
+        )
+      },
+      mental_health: lambda { |calculator|
+        calculator.needs_help_with?("mental_health") && calculator.mental_health_worries != "no"
+      },
+    }.freeze
```

Comment:
This feels a little over complicated. I think the same result could be achieved if `RULES` was simply a list of the existing methods:
```ruby
RULES = %i[
  user_feels_unsafe?
  user_cannot_pay_their_bills?
  user_cannot_get_food?
  user_is_worried_about_going_to_work?
  user_is_unemployed?
  user_needs_somewhere_to_live?
  user_has_mental_health_worries?
]
```
And then `has_results?` would simplify to:
```ruby
def has_results?
  RULES.any? { |rule| __send__(rule) }
end
```
Also each rule method could then be memoized so that they only need to be run once per instance:
```ruby
def user_feels_unsafe?	
  @user_feels_unsafe? ||= needs_help_with?("feeling_unsafe") && feel_safe != "yes"	
end
```
As at the moment each rule is always run twice (to determine `has_results?`) and when called individually.